apps/pie-docs/src/assets/styles/base/_layout.scss

```diff
@@ -142,6 +142,7 @@
 
         > ul + h3,
         > p + h3,
+        > div + h3,
```

Comment:
This is for fixing the margin inconsistency found during design review.

![image](https://github.com/justeattakeaway/pie/assets/448801/16a5bd88-a558-41dc-bc6f-283398667a49)

![image](https://github.com/justeattakeaway/pie/assets/448801/77477161-af47-47be-ba9e-9f67e64bc1a3)

I noticed this CSS change propagates to other pages:

- https://pr1000-docs.pie.design/components/button/overview
- https://pr1000-docs.pie.design/foundations/iconography/overview
- https://pr1000-docs.pie.design/foundations/typography/code